Applications/Spire/Source/Charting/LocalChartModel.cpp

```diff
@@ -42,6 +42,19 @@ QtPromise<std::vector<Candlestick>> LocalChartModel::load(ChartValue first,
   });
 }
 
+void LocalChartModel::store(const std::vector<Candlestick>& candlesticks) {
+  if(candlesticks.empty()) {
```

Comment:
Use ```std::inplace_merge``` to avoid all the copying around.

Reply:
So do I call std::unique() after to avoid duplicates?